crates/trie/sparse/src/state.rs

```diff
@@ -166,6 +200,16 @@ where
         self.storages.insert(address, storage_trie);
     }
 
+    /// Retrieves the storage trie for the given address, creating a new one if it doesn't exist.
+    ///
+    /// This method should always be used to create a storage trie, as it will re-use previously
+    /// allocated and cleared storage tries when possible.
+    fn get_or_create_storage_trie(&mut self, address: B256) -> &mut SparseTrie<S> {
+        self.storages
+            .entry(address)
+            .or_insert_with(|| self.cleared_storages.pop().unwrap_or_default())
+    }
```

Comment:
I see, this will either retrieve an existing trie or reuse an existing, cleared trie